assets/src/edit-story/components/dropDown/index.js

```diff
@@ -0,0 +1,362 @@
+  DropDown.wrapperRef = useRef(null);
+  DropDown.selectRef = useRef();
+  DropDown.arrayOfOptionsRefs = [];
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [focusedValue, setFocusedValue] = useState(null);
+  const focusedIndex = useMemo(
+    () => options.findIndex((item) => item.value === focusedValue),
+    [focusedValue, options]
+  );
+  const activeItem = useMemo(
+    () => options.find((item) => item.value === value),
+    [value, options]
+  );
+  const toggleOptions = useCallback(() => {
+    setIsOpen(false);
+  }, []);
+
+  const handleUpDown = useCallback(
+    ({ key }) => {
+      if (!isOpen) {
+        setIsOpen(true);
+        setFocusedValue(value);
+      } else if (key === 'ArrowUp' && focusedIndex !== 0) {
+        handleMoveFocus(-1);
+      } else if (key === 'ArrowDown' && focusedIndex < options.length - 1) {
+        handleMoveFocus(1);
+      }
+    },
+    [isOpen, value, focusedIndex, options, handleMoveFocus]
+  );
+
+  const handleMoveFocus = useCallback(
+    (offset) => {
+      const findIndex = options.findIndex(
+        (item) => item.value === focusedValue
+      );
+      if (findIndex + offset >= 0 && findIndex + offset < options.length) {
+        setFocusedValue(options[findIndex + offset].value);
+      }
+    },
+    [focusedValue, options]
+  );
+
+  const handleKeyDown = useCallback(
+    ({ keyCode }) => {
+      const searchTerm = searchValue + String.fromCharCode(keyCode);
+      setSearchValue(searchTerm);
+      const searchIndex = options.findIndex((item) =>
+        item.name.toLowerCase().startsWith(searchTerm.toLowerCase())
+      );
+      if (searchIndex >= 0) {
+        setFocusedValue(options[searchIndex].value);
+      }
+      clearSearchValue();
+    },
+    [clearSearchValue, searchValue, options]
+  );
+
+  const handleEnter = useCallback(() => {
+    if (!isOpen) {
+      setIsOpen(true);
+      setFocusedValue(value);
+    } else {
+      handleCurrentValue(focusedValue);
+    }
+  }, [isOpen, focusedValue, handleCurrentValue, value]);
+
+  useFocusOut(DropDown.wrapperRef, toggleOptions);
+  useKeyDownEffect(DropDown.wrapperRef, { key: 'esc' }, toggleOptions, [
+    toggleOptions,
+  ]);
+  useKeyDownEffect(
+    DropDown.wrapperRef,
+    { key: ['up', 'down'], shift: true },
+    handleUpDown,
+    [handleUpDown]
+  );
+  useKeyDownEffect(
+    DropDown.wrapperRef,
+    { key: possibleNormalKeysForSearch, shift: true },
+    handleKeyDown,
+    [handleKeyDown]
+  );
+  useKeyDownEffect(
+    DropDown.wrapperRef,
+    { key: ['space', 'enter'], shift: true },
+    handleEnter,
+    [handleEnter]
+  );
+
+  const clearOptionsRefs = () => {
+    DropDown.arrayOfOptionsRefs = [];
+  };
+
+  useEffect(() => {
+    if (focusedValue) {
+      const findIndex = options.findIndex(
+        (item) => item.value === focusedValue
+      );
+      DropDown.arrayOfOptionsRefs[findIndex].focus();
```

Comment:
What happens if the options list is closed, the select is in focus, and then a search key is typed? This would cause the ref not to exist and an error, wouldn't it?

Reply:
Basically when the option closes, it should also update the focusedvalue to ''.

Reply:
Just tested and this crashes in the Storybook at least when:
- Click on the select to select it
- Click again to close the options
- Click some key, e.g. "u".

Does it work for you?

<img width="1167" alt="Screenshot 2020-03-13 at 16 27 06" src="https://user-images.githubusercontent.com/3294597/76635279-8106a080-6547-11ea-9547-907a3430a368.png">

Reply:
Correct. It has some errors in that case. Missing condition there. Thank you

Reply:
Fixed